src/components/GridContainer/GridContainer.tsx

```diff
@@ -84,36 +105,63 @@ const GridContainer = ({
 const Wrapper = styled.div<{
   $alignContent: ContentOptions;
   $alignItems: ItemsOptions;
-  $columnGap: GapOptions;
-  $gap: GapOptions;
-  $gridAutoColumns: string;
-  $gridAutoFlow: FlowOptions;
-  $gridAutoRows: string;
-  $gridTemplateAreas: string;
-  $gridTemplateColumns: string;
-  $gridTemplateRows: string;
-  $gridTemplate: string;
+  $columnGap?: GapOptions;
+  $gap?: GapOptions;
+  $gridAutoColumns?: string;
+  $gridAutoFlow?: FlowOptions;
+  $gridAutoRows?: string;
+  $gridTemplateAreas?: string;
+  $gridTemplateColumns?: string;
+  $gridTemplateRows?: string;
+  $gridTemplate?: string;
   $inline: boolean;
   $isResponsive: boolean;
   $justifyContent: ContentOptions;
   $justifyItems: ItemsOptions;
-  $rowGap: GapOptions;
+  $rowGap?: GapOptions;
+  $height?: string;
+  $maxHeight?: string;
+  $minHeight?: string;
+  $fillWidth: boolean;
+  $maxWidth?: string;
+  $minWidth?: string;
+  $overflow?: string;
 }>`
   align-items: ${({ $alignItems = "stretch" }) => $alignItems};
   align-content: ${({ $alignContent = "stretch" }) => $alignContent};
-  column-gap: ${({ theme, $columnGap }) => theme.click.gridContainer.gap[$columnGap]};
   display: ${({ $inline }) => ($inline === true ? "inline-grid" : "grid")};
-  gap: ${({ theme, $gap }) => theme.click.gridContainer.gap[$gap]};
-  grid-auto-columns: ${({ $gridAutoColumns }) => $gridAutoColumns || "auto"};
-  grid-auto-flow: ${({ $gridAutoFlow }) => $gridAutoFlow || "auto"};
-  grid-auto-rows: ${({ $gridAutoRows }) => $gridAutoRows || "auto"};
-  grid-template-area: ${({ $gridTemplateAreas }) => $gridTemplateAreas || "auto"};
-  grid-template-columns: ${({ $gridTemplateColumns }) => $gridTemplateColumns || "auto"};
-  grid-template-rows: ${({ $gridTemplateRows }) => $gridTemplateRows || "auto"};
-  grid-template: ${({ $gridTemplate }) => $gridTemplate || "auto"};
+  ${({ $gridAutoColumns }) =>
```

Comment:
what do oyu think of wrigint this as ```
gridAutoColumns && `grid-auto-columns: ${$gridAutoColumns}` 
```

Reply:
and all the ternaries that end up in ""

Reply:
There is a possibility of having false added to the string

Reply:
```
> null && 'foo'
null
> undefined && 'foo'
undefined
```

Reply:
Fixed it